tensorrt_llm/_torch/models/modeling_qwen2vl.py

```
@@ -429,6 +469,272 @@
+                super().__init__()
+                self.original_linear = original_linear
+                
+                # Pre-quantize weights and scaling factors
+                print(f"Pre-quantizing weights for layer with shape {original_linear.weight.shape}")
+                self.weight_fp8, self.weight_scale = self._pre_quantize_weight(original_linear.weight)
+                
+                # Move quantized weights and scaling factors to CPU to save GPU memory
+                self.weight_fp8 = self.weight_fp8.cpu()
+                self.weight_scale = self.weight_scale.cpu()
+                
+                print(f"Pre-quantization completed. Weight FP8 shape: {self.weight_fp8.shape}, Scale shape: {self.weight_scale.shape}")
```

Ruff (0.12.2)

Line too long (135 > 120)

(E501)